recipes/hdf5/meta.yaml

```diff
@@ -0,0 +1,36 @@
+package:
+  name: hdf5
+  version: 1.8.16
+
+source:
+  fn: hdf5-1.8.16.tar.gz
+  url: http://www.hdfgroup.org/ftp/HDF5/current/src/hdf5-1.8.16.tar.gz
+  md5: b8ed9a36ae142317f88b0c7ef4b9c618
+
+
+build:
+  features:
+    - vc9     # [win and py27]
+    - vc10    # [win and py34]
+    - vc14    # [win and py35]
```

Comment:
You do.  These are the actual features.  Python in build only makes the selectors here work (I think).

Importantly, hdf5 should NOT be a python library - it is a lower-level library.  This is why python is only a build-time dependency.  If it were a runtime dependency, it would be tied to that particular version of Python.  Here's a strange use case that would be possible with the former, but not the latter: build Python 2.7 with VS 2015.  We did a very bad job coupling python with visual studio version too strongly.  It is a strong convention, sure, but not a hard requirement.  I hope I have time to correct the way that features are done soon - I will do something more explicit, like what @ukoethe has done in ilastik.